### PR TITLE
fix: redirect issue

### LINKED ISSUE #2446: base pubpub redirect
Talk to Gabe

### PR DESCRIPTION
**Pull request by 3mcd**
Closes #2446 

This PR fixes a couple of minor vulnerabilities:
1. Removes a potential phising attack vector where PubPub would redirect to arbitrary URLs when the pathname began with multiple slashes and were trailed by a single slash, e.g. `http://pubpub.org//google.com/`.
2. Filters sensitive properties from `DepositTarget` models before they are sent to the client when rendering dashboard settings pages (already privileged routes).

I also added `@types/express` so we have typing on our express middleware and route handlers. Still need to type the `wrap` function though.

### CODE REVIEW

File: server/depositTarget/queries.ts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-	service: types.DepositTarget['service'] = 'crossref',
+	includeCredentials = false,
 ): Promise<types.Maybe<types.DepositTarget>> => {
 	return DepositTarget.findOne({
 		where: {
 			communityId,
-			service,
+		},
+		attributes: {
+			exclude: includeCredentials ? [] : ['username', 'password', 'passwordInitVec'],
```
Comment (3mcd) [inline server/depositTarget/queries.ts:18]: A quick fix to ensure we don't send deposit target credentials to browsers viewing dashboard settings pages.

File: server/server.ts [hunk 2]
```diff
@@ -60,6 +61,7 @@ if (process.env.NODE_ENV === 'production') {
 	app.use(Sentry.Handlers.requestHandler({ user: ['id', 'slug'] }));
 	app.use(enforce.HTTPS({ trustProtoHeader: true }));
 }
+app.use(deduplicateSlash());
```
Comment (3mcd) [inline server/server.ts:64]: The main fix. This middleware strips repeating slash characters from request paths.

File: package.json [hunk 3]
```diff
@@ -227,6 +227,7 @@
 		"@storybook/addon-viewport": "^6.4.0",
 		"@storybook/react": "^6.4.0",
 		"@types/enzyme": "^3.10.12",
+		"@types/express": "^4.17.15",
```
Comment (3mcd) [inline package.json:230]: We've got express types now!

**Review by gabestein — APPROVED** [review]
Thanks! Tested and LGTM

**Pull request merged**